pkg/guacanalytics/patchPlanning_test.go

```diff
@@ -0,0 +1,460 @@
+
+					VersionRange: ">=1.19.0",
+
+					DependencyType: model.DependencyTypeIndirect,
+
+					Justification: "test justification one",
+
+					Origin:    "Demo ingestion",
+					Collector: "Demo ingestion",
+				},
+			},
+			{
+				Pkg: &model.PkgInputSpec{
+					Type:      "top3",
+					Namespace: ptrfrom.String("topns3"),
+					Name:      "toppkg3",
+					Version:   ptrfrom.String("1.19.0"),
+				},
+				DepPkg: &model.PkgInputSpec{
+					Type:      "top",
+					Namespace: ptrfrom.String("topns"),
+					Name:      "toppkg",
+					Version:   ptrfrom.String("1.19.0"),
+				},
+				IsDependency: &model.IsDependencyInputSpec{
+					VersionRange:   "<1.19.0",
+					DependencyType: model.DependencyTypeDirect,
+					Justification:  "test justification one",
+					Origin:         "Demo ingestion",
+					Collector:      "Demo ingestion",
+				},
+			},
+			{
+				Pkg: &model.PkgInputSpec{
+					Type:      "deb",
+					Namespace: ptrfrom.String("ubuntu"),
+					Name:      "dpkg",
+					Version:   ptrfrom.String("1.19.0"),
+					Qualifiers: []model.PackageQualifierInputSpec{
+						{Key: "arch", Value: "amd64"},
+					},
+				},
+				DepPkg: &model.PkgInputSpec{
+					Type:      "conan2",
+					Namespace: ptrfrom.String("openssl.org2"),
+					Name:      "openssl2",
+
+					Version: ptrfrom.String("3.0.3"),
+				},
+
+				IsDependency: &model.IsDependencyInputSpec{
+
+					VersionRange: ">=1.19.0",
+
+					DependencyType: model.DependencyTypeDirect,
+
+					Justification: "test justification one",
+
+					Origin:    "Demo ingestion",
+					Collector: "Demo ingestion",
+				},
+			},
+		},
+	}
+)
+
+func ingestIsDependencyTestData(ctx context.Context, client graphql.Client) {
+
+	logger := logging.FromContext(ctx)
+
+	for _, ingest := range isDepTestData.IsDependency {
+
+		_, err := model.IsDependency(context.Background(), client, *ingest.Pkg, *ingest.DepPkg, *ingest.IsDependency)
+
+		if err != nil {
+
+			logger.Errorf("Error in ingesting: %v\n", err)
+		}
+	}
+}
+
+func Test_SearchSubgraphFromVuln(t *testing.T) {
+	server, logger := startTestServer()
+	ctx := logging.WithLogger(context.Background())
+
+	httpClient := http.Client{}
+	gqlclient := graphql.NewClient("http://localhost:9090/query", &httpClient)
+
+	ingestIsDependencyTestData(ctx, gqlclient)
+
+	var pkgIds []string
+
+	for _, dep := range isDepTestData.IsDependency {
+
+		pkgFilter := &model.PkgSpec{
+
+			Type: &dep.Pkg.Type,
+
+			Namespace: dep.Pkg.Namespace,
+			Name:      &dep.Pkg.Name,
+		}
+		pkgResponse, err := model.Packages(ctx, gqlclient, pkgFilter)
+
+		if err != nil {
+			t.Errorf("Error getting id for isDep test case: %s\n", err)
+		}
+
+		pkgIds = append(pkgIds, pkgResponse.Packages[0].Namespaces[0].Names[0].Versions[0].Id)
+	}
+
+	testCases := []struct {
+		startID  string
+		stopID   string
+		maxDepth int
+	}{
+		{
+			// 1: test case with two dependencies at the same depth, no stopID and no limiting maxDepth
+			startID:  pkgIds[0],
```

Comment:
the start node probably should be defined such that it's not dependent on the array index of the graph. It will be nice as a test to be able to read the test case and determine what the test is doing.